braintree/braintree_ios
# [5.x] Add Venmo Enriched Customer Data to BTVenmoRequest

### Summary of changes

This is the same change as PR #974, #1081, and #1075 - these new parameters are required in the Drop-in. Since there is not a plan to update Drop-in to v6 due to the Swift > Obj-C > Swift bug, we need to make the same changes in v5 to port these parameters to the Drop-in.

- Add `BTVenmoLineItem`
- Add new parameters to `BTVenmoRequest`
- Add `isVenmoEnrichedCustomerDataEnabled` to `BTConfiguration+Venmo`
- Update error parsing in `BTGraphQLHTTP`
- Add `billingAddress` and `shippingAddress` to `BTVenmoAccountNonce`
- Add logic to build ECD request in `BTVenmoDriver`
- Add new `BTVenmoDriverErrorTypeEnrichedCustomerDataDisabled` error
- Add ECD to Demo app
- Add new `BTConfiguration+Venmo_Tests` for ECD
- Add new `BTVenmoLineItem_Tests`
- Add error test for alternatively constructed GQL request to `BTGraphQLHTTP_Tests`
- Add additional parameters to `BTVenmoAccountNonce_Tests`
- Add additional ECD tests to `BTVenmoDriver_Tests`
- Add additional `BTVenmoRequest_Tests`

The flow has been tested with and without ECD parameters to ensure no regressions to the standard Venmo flow

### Checklist


### Authors

- @jaxdesmarais
